### PR TITLE
feat(platform/azure): check targetBranch when finding PRs

### PR DESCRIPTION
**Pull request by kroonprins**
## Changes

For the discussion of https://github.com/renovatebot/renovate/discussions/23929

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Note that I have only updated the implementation of `getBranchPr` and `findPr` for the azure platform. Similar things could most probably be done for other platforms, but only for azure devops do I a sufficient setup to test it on a real repository.
I am also not sufficiently aware about all the subtleties in the code to know for sure if all the updated calls to `getBranchPr` and `findPr` would behave as intended, so that should probably be reviewed by someone with more experience in this project.

### CODE REVIEW

File: lib/modules/platform/azure/index.spec.ts [hunk 1]
```diff
@@ -316,6 +316,76 @@ describe('modules/platform/azure/index', () => {
...
+            getPullRequestCommits: jest.fn().mockReturnValue([]),
+          } as any)
+      );
+      const res = await azure.findPr({
+        branchName: 'branch-a',
+        prTitle: 'branch a pr',
+        state: 'open',
+        targetBranch: 'branch-c',
+      });
+      expect(res).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:351]: use `toEqual` or `toMatchObject`

File: lib/modules/platform/azure/index.spec.ts [hunk 2]
```diff
@@ -316,6 +316,76 @@ describe('modules/platform/azure/index', () => {
...
+            getPullRequestCommits: jest.fn().mockReturnValue([]),
+          } as any)
+      );
+      const res = await azure.findPr({
+        branchName: 'branch-a',
+        prTitle: 'branch a pr',
+        state: 'open',
+        targetBranch: 'branch-d',
+      });
+      expect(res).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:386]: see above

File: lib/modules/platform/azure/index.spec.ts (hunk 1 above)
Comment (kroonprins) [reply lib/modules/platform/azure/index.spec.ts:351]: all the other existing tests for `findPr` are using `toMatchSnapshot`.  I either keep it consistent or update them all

Comment (rarkins) [reply lib/modules/platform/azure/index.spec.ts:351]: You can update them all

Comment (kroonprins) [reply lib/modules/platform/azure/index.spec.ts]: updated

File: lib/modules/platform/azure/index.spec.ts [hunk 3]
```diff
@@ -313,7 +355,119 @@ describe('modules/platform/azure/index', () => {
...
+        targetBranch: 'branch-b',
+        targetRefName: 'refs/heads/branch-b',
+        title: 'branch a pr',
+      });
+    });
+
+    it('returns pr if found matches targetBranch', async () => {
+      azureApi.gitApi.mockImplementationOnce(
+        () =>
+          ({
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:378]:
```suggestion
          partial<IGitApi>({
```

File: lib/modules/platform/azure/index.spec.ts [hunk 4]
```diff
@@ -313,7 +355,119 @@ describe('modules/platform/azure/index', () => {
...
+                {
+                  pullRequestId: 2,
+                  sourceRefName: 'refs/heads/branch-a',
+                  targetRefName: 'refs/heads/branch-c',
+                  title: 'branch a pr',
+                  status: PullRequestStatus.Active,
+                },
+              ]),
+            getPullRequestCommits: jest.fn().mockReturnValue([]),
+          } as any)
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:399]:
```suggestion
          })
```

File: lib/modules/platform/azure/index.spec.ts [hunk 5]
```diff
@@ -313,7 +355,119 @@ describe('modules/platform/azure/index', () => {
...
+        targetBranch: 'branch-c',
+        targetRefName: 'refs/heads/branch-c',
+        title: 'branch a pr',
+      });
+    });
+
+    it('returns first pr if found does not match targetBranch', async () => {
+      azureApi.gitApi.mockImplementationOnce(
+        () =>
+          ({
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:427]:
```suggestion
          partial<IGitApi>({
```

File: lib/modules/platform/azure/index.spec.ts [hunk 6]
```diff
@@ -313,7 +355,119 @@ describe('modules/platform/azure/index', () => {
...
+                {
+                  pullRequestId: 2,
+                  sourceRefName: 'refs/heads/branch-a',
+                  targetRefName: 'refs/heads/branch-c',
+                  title: 'branch a pr',
+                  status: PullRequestStatus.Active,
+                },
+              ]),
+            getPullRequestCommits: jest.fn().mockReturnValue([]),
+          } as any)
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:448]:
```suggestion
          })
```

**Review by viceice — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**